### ISSUE TITLE
scsynth documentation: sample formats

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#2764]
The server command reference for  `/b_write`  says:

```
Sample format is one of:
"int8", "int16", "int24", "int32", "float", "double", "mulaw", "alaw"
```

Is that still correct? Doesn't libsndfile offer other possibilities?

**Comment by vivid-synth** [comment]
scsynth and supernova have produced corrupted .flac and .ogg files for me. If nobody knows the reason for this I can dig up a repro case.

**Comment by bagong** [comment]
I am not sure that writing flac and ogg is supported at all. From a pragmatic perspective the logic being that an as sensitive operation like writing to a file while SC is working wouldn't usually be burdened by a potentially costly process like compressing a sound file. But it's open to inspection, of course, work on enhancing and cleaning up soundfile integration is highly welcome. Quite a few formats don't work. Would be quite nice actually to have a tool that can write these historic file formats. Also the code integrating libsndfile is quite messy as it was changed hastily at least twice for OSX. There was also a problem once that libsndfile was required in the Qt-part, although in theory it should be possible to build SC without libsndfile. But that was fixed, I think. Current state if I remember correctly: you can write at least one of the uncompressed formats that allow for large files (more than 2 or 4 GB) on each platform. Additionally you can read various popular compressed formats.

**Comment by jamshark70** [comment]
I'd tend to agree with @bagong that it's risky to write a compressed audio format using DiskOut -- so much so that I'd be content to document it as an "at your own risk" activity.

Writing a `SoundFile` in a compressed format should be expected to work, though, and perhaps even `/b_write` (which is deferred to a non-realtime thread and thus shouldn't interfere with RT audio processing).

**Comment by vivid-synth** [comment]
- It would be great to have writing of compressed formats, for NRT
  - Regarding writing .raw but not reading: write-from-SC-then-reread is common; it would be great to support reading of whichever formats we can write

**Comment by telephon** [comment]
I'm not sure if the compression process that happens for writing (or reading) flac files would really make any difference compared to any other audio task.

**Comment by bagong** [comment]
Well, would be nice to have the possibility anyways, no question... I am not suggesting to avoid offering it. Was just wondering why it wasn't available... Flac and ogg support had been available in the past, but it broke when libsndfile 1.0.24 - universal build - was bundled and statically linked back in the times when Mac moved to Intel (the statically linked lib didn't contain the flac and ogg libraries). The support was recovered for 3.7.1, but as it turned out no writing...

**Comment by jamshark70** [comment]
It's exceedingly rare for professional-grade software synths and samplers to support compressed formats natively -- granted, I haven't checked recent versions of Kontakt but I'd be surprised if it allowed you to drop FLAC files into an instrument grid without conversion to a WAV on disk. (Loading large Kontakt patches can be slow enough without adding file decompression to the process.)

I'd say, nice to have, but nobody expects it of state-of-the-art commercial softsynths, and this point will hardly be SC's great distinction in the marketplace. Go ahead if it's easy but we have tons of issues that are more important.

**Comment by bagong** [comment]
I agree, @jamshark70 . As the smell of 3.9 is getting stronger it might be good get a good idea of the top priorities for 3.9. Quite a bit has changed since we initially thought about it. For example the replacement of QtWebkit with QtWebengine was thought to be a necessity, but it looks like alternative QtWebkits have emerged , that give us some time now... Oth some fascinating potential is getting in reach like Unicode support, something nobody dared dared to dream about when we were planning. Maybe it would be good to get a dedicated discussion going what can or has to be done for 3.9.
